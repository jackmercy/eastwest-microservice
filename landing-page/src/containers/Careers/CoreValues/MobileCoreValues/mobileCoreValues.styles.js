import { createUseStyles } from 'react-jss';

import { rem } from '../../../../utils/jss';
import COLORS from '../../../../constants/COLORS';
import BREAKPOINTS from '../../../../constants/BREAKPOINTS';

export default createUseStyles({
  detailsWrapper: {
    display: 'inline-flex',
    boxSizing: 'border-box',
    alignItems: 'flex-start',
    justifyContent: 'space-between',
    marginBottom: rem(64),
  },
  details: { width: '100%', maxWidth: rem(528) },
  detailsSubTitle: {
    margin: 0,
    fontWeight: 600,
    fontSize: rem(12),
    lineHeight: rem(16),
    color: COLORS.primary,
  },
  sectionTitle: {},
  detailsDescription: {
    fontSize: rem(24),
    lineHeight: rem(40),
    marginTop: rem(24),
    marginBottom: 0,
  },
  image: {
    display: 'block',
    boxSizing: 'border-box',
    marginLeft: rem(64),
    flex: 'none',
    maxHeight: rem(426),
    minWidth: rem(200),
    maxWidth: rem(300),
  },
  listKeys: {
    display: 'flex',
    justifyContent: 'space-between',
    marginTop: rem(84),
  },
  listKeysItem: {
    display: 'flex',
    flex: 1,
    boxSizing: 'border-box',
    cursor: 'pointer',
    '&:nth-child(2), &:nth-child(3)': { padding: [0, rem(8)] },
    '&.active': { color: COLORS.primary },
  },
  listKeysNumber: {
    marginTop: rem(-5),
    marginBottom: 0,
    fontWeight: 900,
    fontSize: rem(32),
    lineHeight: rem(40),
  },
  listKeysLabel: {
    margin: [0, 0, 0, rem(20)],
    fontWeight: 600,
    fontSize: rem(24),
    lineHeight: rem(32),
  },

  [`@media only screen and (max-width: ${BREAKPOINTS.maxSm}px)`]: {
    detailsWrapper: { display: 'block' },
    details: { textAlign: 'center' },
    detailsDescription: {
      fontSize: rem(16),
      lineHeight: rem(24),
      marginTop: 0,
    },
    sectionTitle: { fontSize: rem(32), marginBottom: rem(16) },
    image: { margin: [rem(32), 'auto', 0], maxWidth: rem(275) },
    listKeys: { display: 'none' },
    listKeysNumber: {},
    listKeysLabel: {},
  },
});
