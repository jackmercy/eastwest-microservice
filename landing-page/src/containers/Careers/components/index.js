export { default as SectionTitle } from './SectionTitle';
export { default as MenuScrollableLayout } from './MenuScrollableLayout';
