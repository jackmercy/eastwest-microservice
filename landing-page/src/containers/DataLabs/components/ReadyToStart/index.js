import dynamic from 'next/dynamic';

export default dynamic(() => import('./ReadyToStart'), { ssr: false });
