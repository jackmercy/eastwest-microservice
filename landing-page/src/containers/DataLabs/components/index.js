export { default as Teaser } from './Teaser';
export { default as JoinUs } from './JoinUs';
export { default as Become } from './Become';
export { default as HowItWorks } from './HowItWorks';
export { default as ReadyToStart } from './ReadyToStart';
