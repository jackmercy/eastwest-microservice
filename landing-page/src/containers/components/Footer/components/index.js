export { default as Register } from './Register';
export { default as CopyRight } from './CopyRight';
export { default as SummaryInfo } from './SummaryInfo';
export { default as SectionTitle } from './SectionTitle';
export { default as SwitchLanguages } from './SwitchLanguages';
