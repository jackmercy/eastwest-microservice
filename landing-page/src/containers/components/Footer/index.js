import dynamic from 'next/dynamic';

export default dynamic(() => import('./Footer'), { ssr: false });
