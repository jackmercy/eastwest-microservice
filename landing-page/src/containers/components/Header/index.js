import dynamic from 'next/dynamic';

export default dynamic(() => import('./Header'), { ssr: false });
