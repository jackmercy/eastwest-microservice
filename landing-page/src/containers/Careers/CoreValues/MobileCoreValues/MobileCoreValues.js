import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { useIntl } from 'react-intl';

import { SectionTitle } from '../../components';

import useStyles from './mobileCoreValues.styles';

const MobileCoreValues = ({ data }) => {
  const intl = useIntl();
  const classes = useStyles();

  return data.map((item) => (
    <div
      key={item.key}
      className={classes.detailsWrapper}
      data-test="mobile-core-value-item"
    >
      <div className={classes.details}>
        <p
          className={classes.detailsSubTitle}
          data-test="mobile-core-value-sub-title"
        >
          {intl.formatMessage({ id: 'CAREERS.MENU.02' })}
        </p>
        <SectionTitle
          id={`core-value-${item.key}`}
          translateId={item.titleTranslationId}
          className={classes.sectionTitle}
        />
        <p
          className={classes.detailsDescription}
          data-test="mobile-core-value-description"
        >
          {item.description}
        </p>
      </div>
      <img
        alt={item.title}
        src={item.image}
        className={classes.image}
        srcSet={item.srcSet}
        data-test="mobile-core-value-image"
      />
    </div>
  ));
};
MobileCoreValues.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      image: PropTypes.string.isRequired,
      srcSet: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default memo(MobileCoreValues);
