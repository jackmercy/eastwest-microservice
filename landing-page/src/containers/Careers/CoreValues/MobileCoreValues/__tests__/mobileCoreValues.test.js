import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { SectionTitle } from '../../../components';
import MobileCoreValues from '../MobileCoreValues';

describe('Testing Careers > CoreValues > MobileCoreValues', () => {
  let wrapper;
  const props = {
    data: [
      {
        key: 'key-01',
        title: 'title 01',
        titleTranslationId: 'CAREERS.CORE_VALUE.01.TITLE',
        description: 'descriptipn 01',
        image: 'image-01.png',
        srcSet: 'image-01.png 500w',
      },
      {
        key: 'key-02',
        title: 'title 01',
        titleTranslationId: 'CAREERS.CORE_VALUE.02.TITLE',
        description: 'description 02',
        image: 'image-02.png',
        srcSet: 'image-02.png 500w',
      },
    ],
  };

  beforeAll(() => {
    wrapper = mount(<MobileCoreValues {...props} />);
  });

  it('should render match snapshot', () => {
    const wrapper = shallow(<MobileCoreValues {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have number items equal with data input', () => {
    expect(wrapper.find('[data-test="mobile-core-value-item"]')).toHaveLength(
      props.data.length,
    );
  });

  it('each item should have a sub title', () => {
    expect(
      wrapper.find('[data-test="mobile-core-value-sub-title"]'),
    ).toHaveLength(props.data.length);
  });

  it('each item should have SectionTitle', () => {
    expect(wrapper.find(SectionTitle)).toHaveLength(props.data.length);
  });

  it('each item should have a description', () => {
    expect(
      wrapper.find('[data-test="mobile-core-value-description"]'),
    ).toHaveLength(props.data.length);
  });

  it('each item should have a cover image', () => {
    expect(wrapper.find('[data-test="mobile-core-value-image"]')).toHaveLength(
      props.data.length,
    );
  });
});
