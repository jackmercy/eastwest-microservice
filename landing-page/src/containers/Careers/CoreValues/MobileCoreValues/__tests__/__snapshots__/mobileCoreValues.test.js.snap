// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Careers > CoreValues > MobileCoreValues should render match snapshot 1`] = `
Array [
  <div
    className="detailsWrapper-0-2-1"
    data-test="mobile-core-value-item"
    key="key-01"
  >
    <div
      className="details-0-2-2"
    >
      <p
        className="detailsSubTitle-0-2-3"
        data-test="mobile-core-value-sub-title"
      >
        Core values
      </p>
      <Memo(SectionTitle)
        className="sectionTitle-0-2-4"
        id="core-value-key-01"
        translateId="CAREERS.CORE_VALUE.01.TITLE"
      />
      <p
        className="detailsDescription-0-2-5"
        data-test="mobile-core-value-description"
      >
        descriptipn 01
      </p>
    </div>
    <img
      alt="title 01"
      className="image-0-2-6"
      data-test="mobile-core-value-image"
      src="image-01.png"
      srcSet="image-01.png 500w"
    />
  </div>,
  <div
    className="detailsWrapper-0-2-1"
    data-test="mobile-core-value-item"
    key="key-02"
  >
    <div
      className="details-0-2-2"
    >
      <p
        className="detailsSubTitle-0-2-3"
        data-test="mobile-core-value-sub-title"
      >
        Core values
      </p>
      <Memo(SectionTitle)
        className="sectionTitle-0-2-4"
        id="core-value-key-02"
        translateId="CAREERS.CORE_VALUE.02.TITLE"
      />
      <p
        className="detailsDescription-0-2-5"
        data-test="mobile-core-value-description"
      >
        description 02
      </p>
    </div>
    <img
      alt="title 01"
      className="image-0-2-6"
      data-test="mobile-core-value-image"
      src="image-02.png"
      srcSet="image-02.png 500w"
    />
  </div>,
]
`;
